llvm/test/Analysis/ScalarEvolution/udiv-of-x-xsmaxone-fold.ll

```diff
@@ -0,0 +1,78 @@
+; NOTE: Assertions have been autogenerated by utils/update_analyze_test_checks.py
+; RUN: opt -disable-output "-passes=print<scalar-evolution>" < %s 2>&1 | FileCheck %s
+
+@g_var = external local_unnamed_addr global i32, align 4
+
+define void @test(i32 noundef %x) {
```

Comment:
This test looks way too complicated.

Reply:
Updated by letting LV perform the actual transformation, think this should make sense.

Reply:
Would be great to avoid having a SCEV analysis test depend on running LV. Is there anything that makes it difficult to write a simple IR test that contains an `(-1 + (1 smax %x))<nsw> /u %x)` expression?

Reply:
Exceptionally, I think the main point here was to demonstrate how LV itself generates that expression from a relatively simple loop, instead of manually handcrafting ourselves an IR with a bit of an unnatural / artificial expression; ergo, showing where the expression actually comes from. I understand the concern though, I'm happy to change the IR if we don't want to have this.

Reply:
Test rewritten so as to avoid the dependence with LV.